### PR TITLE
Setup the PostgreSQL Database

### PR DESCRIPTION
**Pull request by bryanansong**

### CODE REVIEW

File: backend/prisma/schema.prisma [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+model User {
+  id                Int                @id @default(autoincrement())
+  username          String             @unique
+  email             String             @unique
+  password          String
+  workoutTemplates  WorkoutTemplate[]
+  workoutSessions   WorkoutSession[]
+  following  Follow[]           @relation("UserFollowing")
```
Comment (m5matthew) [inline backend/prisma/schema.prisma]: Question: Does setting a relation in prisma automatically set an index on the table? Eg. if I want to query all the users who user1 follows, is it O(1) or O(N).

File: backend/prisma/schema.prisma [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+model SetRecord {
+  id               Int            @id @default(autoincrement())
+  workoutSessionId Int
+  exerciseId       Int
+  workoutSession   WorkoutSession @relation(fields: [workoutSessionId], references: [id])
+  exercise         Exercise       @relation(fields: [exerciseId], references: [id])
+  reps             Int
+  weight           Float
```
Comment (m5matthew) [inline backend/prisma/schema.prisma:83]: unit will be in lbs? or kg? (might be good to add a comment similar to duration above)

Comment (bryanansong) [reply backend/prisma/schema.prisma:83]: I was thinking of using lbs. I will add a comment for that field

File: backend/prisma/schema.prisma (hunk 1 above)
Comment (bryanansong) [reply backend/prisma/schema.prisma]: I don't know about this, however, I don't think it sets an index so your example would probably be an O(N) operation.

Comment (m5matthew) [reply backend/prisma/schema.prisma]: sounds good. as a stretch goal if we get to it, or if we notice the app is loading slowly, I would suggest us adding an index to this table to make it scale a bit better

**Pull request merged**